enhancements/proposed/20200501-json-output/start-json-output.md

```diff
@@ -57,6 +70,53 @@ If minikube fails, and an actionable error message exists, the following JSON wi
 ```
 
 ### Stdout
+If `--output json` is specified, each step of `minikube start` will be output as JSON:
+
+```json
+{
+  "Name": "Selecting Driver",
+  "Message": "✨  Using the hyperkit driver based on user configuration\n",
+  "TotalSteps": 9,
+  "CurrentStep": 2,
+  "Type": "Log"
+}
+```
+
+and each update on a downloaded artifact will be output as:
+
+```json
+{
+  "Type": "Download",
+  "Artifact": "preload.tar.gz",
+  "Progress": "10%"
+}
+```
+
+`minikube start` output would look something like this:
+
+```
+$ minikube start
+{"Name":"Minikube Version","Message":"😄  minikube v1.10.0-beta.2 on Darwin 10.14.6\n","TotalSteps":9,"CurrentStep":1, "Type":"Log"}
+{"Name":"Selecting Driver","Message":"✨  Using the hyperkit driver based on user configuration\n","TotalSteps":9,"CurrentStep":2,"Type":"Log"}
+{"Name":"Starting Control Plane","Message":"👍  Starting node minikube in cluster minikube\n","TotalSteps":9,"CurrentStep":3,"Type":"Log"}
+{"Name":"Download Necessary Artifacts","Message":"💾  Downloading Kubernetes v1.18.1 preload ...\n","TotalSteps":9,"CurrentStep":4,"Type":"Log"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "10%"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "61%"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "73%"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "87%"}
+  {"Type":"Download", "Artifact":"preload.tar.gz", "Progress": "100%"}
+{"Name":"Creating Node","Message":"🔥  Creating hyperkit VM (CPUs=2, Memory=6000MB, Disk=20000MB) ...\n","TotalSteps":9,"CurrentStep":5,"Type":"Log"}
+{"Name":"Preparing Kubernetes","Message":"🐳  Preparing Kubernetes v1.18.1 on Docker 19.03.8 ...\n","TotalSteps":9,"CurrentStep":6,"Type":"Log"}
+{"Name":"Verifying Kubernetes","Message":"🔎  Verifying Kubernetes components...\n","TotalSteps":9,"CurrentStep":7,"Type":"Log"}
+{"Name":"Enabling Addons","Message":"🌟  Enabled addons: default-storageclass, storage-provisioner\n","TotalSteps":9,"CurrentStep":7,"Type":"Log"}
+{"Name":"Done","Message":"🏄  Done! kubectl is now configured to use \"minikube\"\n","TotalSteps":9,"CurrentStep":9,"Type":"Log"}
+```
+
```

Comment:
Could you give an example of an error message?

Reply:
Just updated the proposal so there should be an error at the bottom now